pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -516,8 +555,12 @@ func (a *Actuator) Update(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 	// but instance could be deleted between the two calls.
 	if len(instances) == 0 {
 		mLog.Warnf("attempted to update machine but no instances found")
+		err := a.updateProviderID(machine, nil, mLog)
```

Comment:
Given the provider ID does not change very often, it's fine to update entire machine object followed by updating just its status.